src/views/Auth/Auth.jsx

```diff
@@ -0,0 +1,71 @@
+import { useState} from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useUser } from '../../context/UserContext';
+import { signUpUser } from '../../services/user';
+import authStyle from './Auth.css'
+
+export default function Auth() {
+  const { setUser, login } = useUser()
+
+  const history = useHistory();
+
+  const [email, setEmail] = useState('');
+  const [password, setPassword] = useState('');
+  const [error, setError] = useState('');
+  const [type, setType] = useState('sign-in');
+
+  const handleSubmit = async (event) => {
+    // add preventDefault to prevent 1996 behavior in the form
+    // event.preventDefault();
+    try{
+    if (type === 'sign-in') {
+      event.preventDefault();
+      await login(email, password);
+      // not setting user here because login function does so in context
+      history.replace('/');
+    } else {
+      event.preventDefault();
+      const resp = await signUpUser(email, password);
+      setUser(resp.email);
+      history.replace('/');
```

Comment:
In order to get the signup method working, we will need to remove .email from user and you should be redirected correctly.